examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+      "Using local simulator:  Lightning Qubit PennyLane plugin\n",
+      "\n",
+      "Gradient descent optimizer:\n",
+      "Step 0:  loss = -2.55\n",
+      "Step 20:  loss = -3.17\n",
+      "Step 40:  loss = -3.23\n",
+      "Step 60:  loss = -3.21\n",
+      "Step 80:  loss = -3.21\n",
+      "\n",
+      "Quantum natural gradient optimizer:\n",
+      "Step 0:  loss = -2.55\n",
+      "Step 20:  loss = -3.20\n",
+      "Step 40:  loss = -3.25\n",
+      "Step 60:  loss = -3.24\n",
+      "Step 80:  loss = -3.20\n",
+      "\n",
+      "QN-SPSA optimizer:\n",
+      "Trace 0:\n",
+      "Step 0:  loss = -2.54\n",
+      "Step 20:  loss = -2.72\n",
+      "Step 40:  loss = -3.06\n",
+      "Step 60:  loss = -3.11\n",
+      "Step 80:  loss = -3.15\n",
+      "Trace 1:\n",
+      "Step 0:  loss = -2.56\n",
+      "Step 20:  loss = -2.91\n",
+      "Step 40:  loss = -3.06\n",
+      "Step 60:  loss = -3.12\n",
+      "Step 80:  loss = -3.54\n",
+      "\n",
+      "SPSA optimizer:\n",
+      "Trace 0:\n",
+      "Step 0:  loss = -2.62\n",
+      "Step 20:  loss = -3.14\n",
+      "Step 40:  loss = -3.06\n",
+      "Step 60:  loss = -3.16\n",
+      "Step 80:  loss = -3.07\n",
+      "Trace 1:\n",
+      "Step 0:  loss = -2.55\n",
+      "Step 20:  loss = -3.14\n",
+      "Step 40:  loss = -2.99\n",
+      "Step 60:  loss = -3.18\n",
+      "Step 80:  loss = -3.19\n",
+      "Training Successful!!\n",
+      "Code Run Finished\n",
+      "CPU times: user 315 ms, sys: 111 ms, total: 427 ms\n",
+      "Wall time: 1min 31s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "n_qubits = 4\n",
+    "edges = 5\n",
+    "depth = 2\n",
+    "shots = 1000\n",
+    "seed = 1\n",
+    "load_init_config = False\n",
+    "max_iter = 100\n",
+    "spsa_repeats = 2\n",
+    "\n",
+    "hyperparameters = {\n",
+    "    \"n_qubits\": n_qubits,\n",
+    "    \"edges\": edges,\n",
+    "    \"depth\": depth,\n",
+    "    \"shots\": shots,\n",
+    "    \"seed\": seed,\n",
+    "    \"load_init_config\": load_init_config,\n",
+    "    \"max_iter\": max_iter,\n",
+    "    \"spsa_repeats\": spsa_repeats,\n",
+    "}\n",
+    "\n",
+    "# This cell should take about 2 min for the first time, \n",
+    "# and about 30 seconds afterward.\n",
+    "\n",
+    "job = LocalQuantumJob.create(\n",
+    "    device=\"local:pennylane/lightning.qubit\",\n",
+    "    source_module=\"source_scripts\",\n",
+    "    entry_point=\"source_scripts.benchmark_qaoa_converge_speed\",\n",
+    "    job_name=\"local-embedded-simulation-\" + str(int(time.time())),\n",
+    "    hyperparameters=hyperparameters,\n",
+    "    image_uri=image_uri,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e22e7e0",
+   "metadata": {},
+   "source": [
+    "Note that each local Job will create a folder under the current path storing the log and result. This is to help users to retrieve info from past runs. Feel free to delete them as needed."
```

Comment:
Re: "This is to help users to retrieve info from past runs": Let's make this more personal: "This feature would be helpful if I needed to understand the details of my past runs."